massgov/mayflower-patternlab
# Dp 7718: [Search] Add support for Description text to Callout Link Molecule

The search application requires, based on the promoted results designs, a variant of the Callout Link molecule that includes description text.

## Description
Add description support for the Callout Link Molecule in Mayflower. The aim of adding description support to the callout link is to be able to meet the requirements of the search application for [promoted results](https://projects.invisionapp.com/d/#/console/13018647/272994955/preview).
<img width="415" alt="screen shot 2018-02-02 at 12 54 56 pm" src="https://user-images.githubusercontent.com/9359261/35747493-6af7aa2a-0818-11e8-9017-c8411d4fa973.png">

## Related Issue / Ticket

- [DP-7718](https://jira.state.ma.us/browse/DP-7718)

## Steps to Test
1. Compile Mayflower branch and view in browser
2. Check template at Molecules > Callout Link as Description
3. Confirm that the Callout Link molecule displays with a description as shown in the screenshot below. Further, confirm that the description does not display at a mobile screen size, as in the screenshot shown below. Confirm that the readme links to the Callout Link molecule and provides information on how to create this variant
4. Check template at Molecules > Callout Link
5. Confirm that the json variable includes description as optional. Confirm that that twig template include conditional rendering of description text logic, if it exists. Confirm that the readme mentions the description variant.
 

## Screenshots
![callout-link-as-description](https://user-images.githubusercontent.com/9359261/35747332-e86982ea-0817-11e8-810e-5e7f2e791998.gif)


## Additional Notes:
 
@jesconstantine can you double check this PR to make sure it is set, my first commit to Mayflower code so want to be sure it looks good and is up to par/standards.

#### Impacted Areas in Application


* 

#### @TODO


*
